city/crud.py

```diff
@@ -0,0 +1,199 @@
+import json
+from typing import Any
+from datetime import datetime
+
+from fastapi import HTTPException
+from requests import Response, request
+from sqlalchemy import delete, insert, select, update
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from city import models, schemas
+from settings import settings
+
+
+async def get_cities_list(db: AsyncSession) -> list[models.DBCity]:
+    query = select(models.DBCity)
+    city_list = await db.execute(query)
+
+    return [city[0] for city in city_list.all()]
+
+
+async def get_city(city_id: int, db: AsyncSession) -> models.DBCity:
+    query = select(models.DBCity).filter(models.DBCity.id == city_id)
+
+    db_city = await db.execute(query)
+    db_city = db_city.first()
+
+    if db_city is None:
+        raise HTTPException(
+            status_code=404,
+            detail=f"City with id {city_id} not found",
+        )
+
+    return db_city[0]
+
+
+async def create_city(
+        city: schemas.CityCreate,
+        db: AsyncSession,
+) -> dict[str, Any]:
+    query = insert(models.DBCity).values(
+        name=city.name,
+        additional_info=city.additional_info,
+    )
+
+    result = await db.execute(query)
+    await db.commit()
+
+    response = {**city.model_dump(), "id": result.lastrowid}
+
+    return response
+
+
+async def update_city(
+    city: schemas.CityCreate,
+    city_id: int,
+    db: AsyncSession
+) -> dict[str, Any]:
+    query = (
+        update(models.DBCity).where(models.DBCity.id == city_id).values(
+            name=city.name,
+            additional_info=city.additional_info,
+        )
+    )
+
+    await db.execute(query)
+    await db.commit()
+
+    response = {**city.model_dump(), "id": city_id}
+
+    return response
+
+
+async def delete_city(city_id: int, db: AsyncSession) -> dict[str, str]:
+    query = delete(models.DBCity).where(models.DBCity.id == city_id)
+
+    await db.execute(query)
+    await db.commit()
+
+    return {"Message": "Item was deleted"}
+
+
+async def make_weather_api_get_request(city_name: str) -> Response:
+    url = (
+        f"{settings.temperature_api_url}"
+        f"?key={settings.temperature_api_key}&q={city_name}"
+    )
+
+    response = request(url=url, method="GET")
```

Comment:
The `request` function from the `requests` library is synchronous. Consider using an asynchronous HTTP client like `httpx` to make non-blocking requests in an async function.